inventory_reporting/reports/inventory_report.py

```diff
@@ -0,0 +1,334 @@
+        if html_content:
+            soup = BeautifulSoup(html_content, "html.parser")
+            return soup.get_text()
+        return False
+
+    def generate_valuation_report(self, workbook, wizard):
+        categories = [
+            "Harness",
+            "PCBA",
+            "PCB",
+            "Electrical",
+            "Component",
+            "Mechanical",
+            "Consumable",
+        ]
+
+        for _i, category in enumerate(categories):
+            ws = workbook.add_worksheet(category)
+
+            # Write the header
+            headers = [
+                _("Product Name"),
+                _("Quantity"),
+                _("Unit of Measurement"),
+                _("Unit Price"),
+                _("Total Value"),
+                _("Last Purchase Date"),
+            ]
+            for col, header in enumerate(headers):
+                ws.write(0, col, header)
+
+            # Fetch the valuation layers for the product category and date range
+            valuation_obj = self.env["stock.valuation.layer"]
+            valuations = valuation_obj.search(
+                [
+                    ("product_id.active", "=", True),
+                    ("product_id.categ_id.name", "=", category),
+                    ("accounting_date", "<=", wizard.date_end),
+                ]
+            )
+
+            # Group valuations by product and aggregate the quantities and total values
+            grouped_valuations = {}
+            for valuation in valuations:
+                product_id = valuation.product_id.id
+                if product_id not in grouped_valuations:
+                    grouped_valuations[product_id] = {
+                        "product": valuation.product_id,
+                        "quantity": 0,
+                        "unit_cost": 0,
+                        "value": 0,
+                    }
+                grouped_valuations[product_id]["quantity"] += valuation.quantity
+                grouped_valuations[product_id]["unit_cost"] += valuation.unit_cost
+                grouped_valuations[product_id]["value"] += valuation.value
+
+            # Write the aggregated data
+            row = 1
+            for valuation_data in grouped_valuations.values():
+                product = valuation_data["product"]
+                ws.write(row, 0, product.name)
+                ws.write(row, 1, valuation_data["quantity"])
+                ws.write(row, 2, product.uom_id.name)
+                ws.write(row, 3, valuation_data["value"])
+                ws.write(row, 4, valuation_data["value"])
+
+                # Convert the date to the desired format (YYYY-MM-DD)
+                last_purchase_date = fields.Date.from_string(product.last_purchase_date)
+                if last_purchase_date:
+                    ws.write(row, 5, last_purchase_date.strftime("%Y-%m-%d"))
+                row += 1
+
+    def generate_storable_report(self, workbook, wizard):
+        categories = [
```

Comment:
I think all this should be made configurable in the product category.  We may follow up on this later as a maintenance task if it cannot be handled quickly.